File: 준영/LV2/행렬의 곱셈.js
function solution(arr1, arr2) {
  var N = arr1.length
  var M = arr1[0].length
  var N1 = arr2.length
  var M1 = arr2[0].length
  
  var answer = Array.from(Array(N), () => Array(M1).fill(0))

  for (let i = 0; i < N; i++) {
    for (let j = 0; j < M1; j++) {
      var temp = 0
      for (let k = 0; k < M; k++) {
        temp += arr1[i][k] * arr2[k][j] 
      }
      answer[i][j] = temp
    }
  }
  return answer
}

File: 준영/프로그래머스/LV2/행렬의 곱셈.py
def solution(arr1, arr2):
    N = len(arr1)
    M = len(arr1[0])
    N1 = len(arr2)
    M1 = len(arr2[0])
    answer = [[0 for _ in range(M1)] for _ in range(N)]

    for i in range(N):
        for j in range(M1):
            temp = 0
            for k in range(M):
                temp += arr1[i][k] * arr2[k][j]
            answer[i][j] = temp


    return answer

arr1 = [[1, 4], [3, 2], [4, 1]]	
arr2 = [[3, 3], [3, 3]]
solution(arr1, arr2)